lean/deps/galois_stdlib/src/galois/category/combinators.lean

```diff
@@ -0,0 +1,11 @@
+/-
+This defines a few combinators for monads that aren't in the standard library.
+-/
+
+universes u v
+
+def pwhen {m : Type u → Type v} [monad m] (c : Prop) [h : decidable c] (t : m punit) : m punit :=
```

Comment:
I know these are very simple/straight forward definitions but it would be a good habit to still write some docs for them.